src/sql/workbench/common/editor/query/queryEditorInput.ts

```diff
@@ -346,6 +321,28 @@ export abstract class QueryEditorInput extends EditorInput implements IConnectab
 			}
 		}
 		this._onDidChangeLabel.fire();
+
+		this.contextualizeEditorForCopilot();
+	}
+
+	private contextualizeEditorForCopilot(): void {
+		// Don't need to take any actions if contextualization is not enabled and can return
+		if (!this.configurationService.getValue<IQueryEditorConfiguration>('queryEditor').githubCopilotContextualizationEnabled) {
+			return;
+		}
+
+		this.serverContextualizationService.getServerContextualization(this.uri)
+			.then(async getServerContextualizationResult => {
+				if (getServerContextualizationResult.context) {
+					await this.serverContextualizationService.sendServerContextualizationToCopilot(getServerContextualizationResult.context);
+				}
+				else {
```

Comment:
I think it would be better, but I wasn't sure if generating and reading context was considered doing too much in a single request endpoint, which is why I kept them separate.

I can consolidate the two requests into a single one, so that this new request can read the context or generate context if there isn't any available.

Reply:
Sounds good, I think consolidating into one makes sense to keep things simpler. Then you can easily add something like a "forceRefresh" param as well later on to force regenerating the context if that's something you want (probably useful for users to be able to control like how we have a refresh intellisense command)